usersession/agent/session_agent.go

```diff
@@ -227,6 +266,7 @@ func (s *SessionAgent) shutdownServerOnKill() error {
 	// logic as well.
 	s.listener.Close()
 	systemd.SdNotify("STOPPING=1")
```

Comment:
Does it matter when we tell systemd that we are stopping?
In addition putting the sd_notify call in the middle of other close calls is surprising.

Reply:
conceptually at least is nice to do it after we have stopped accepting requests

Reply:
Here's what the thing systemd docs say:
```
       STOPPING=1
           Tells the service manager that the service is beginning its
           shutdown. This is useful to allow the service manager to track the
           service's internal state, and present it to the user.
```

So it makes sense to enter it at the beginning of the shutdown process.  Conceptually we could put it before closing the sockets too.